tests/test_utils/postgres_system_helpers.py

```diff
@@ -0,0 +1,97 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from contextlib import contextmanager
+from typing import List
+
+import pytest
+
+from tests.test_utils.system_tests_class import SystemTest
+from tests.utils.logging_command_executor import get_executor
+
+
+@contextmanager
+def provide_postgres_schema(name):
+    PostgresSystemTest.create_schema(name)
+    yield name
+    PostgresSystemTest.delete_schema(name)
+
+
+@contextmanager
+def provide_postgres_table(schema_name, table_name, columns=None):
+    PostgresSystemTest.create_table(schema_name, table_name, columns)
+    yield
+    PostgresSystemTest.delete_table(schema_name, table_name)
+
+
+@pytest.mark.system("postgres")
+class PostgresSystemTest(SystemTest):
+
+    @classmethod
+    def execute_with_ctx(cls, cmd: str):
+        """
+        Executes command with postgres (psql cli) context.
+        """
+        executor = get_executor()
+        from os import getenv
+        connection_string = getenv("AIRFLOW_CONN_POSTGRES_DEFAULT")
+        auth = ["psql", connection_string]
+        executor.execute_cmd(cmd=auth + ["-c", cmd])
```

Comment:
Again, why are we shelling out rather than using DbAPiHook/psycopg2?

Reply:
Same as above.